runtime/lua/vim/lsp/buf.lua

```diff
@@ -57,43 +57,43 @@ end
 --- Displays hover information about the symbol under the cursor in a floating
 --- window. Calling the function twice will jump into the floating window.
 function M.hover()
-  local params = util.make_position_params()
+  local params = util.make_position_params() -- TODO I don't know how to get offset_encoding in this file
```

Comment:
@bfredl help wantted.

Reply:
`vim.lsp.buf_get_clients()` ?

Reply:
these all apply to "current buffer" so I guess the same approach as in lsp.omnifunc could be used. Note that `0` means "current buffer".

```lua
local bufclients = vim.lsp.buf_get_clients(0)
local offset_encoding = bufclients[0].offset_encoding
local params = util.make_position_params(offset_encoding)
```

(this is just for illustration, this code should not be copy-pasted everywhere...)

Reply:
@mjlbach look here. Currently, we generate param before `request`, but different clients should get different position due to they have own `offset_encoding`, so I let request take a function to generate param for each client.

Reply:
I'm aware of what you've done. I'm saying at minimum don't rename the inputs because you changed something up the call stack.

I'm also not convinced this is the correct approach. I'd rather this PR stuck to the original scope.